resources/domain_join.rb

```diff
@@ -120,7 +120,78 @@
   end
 end
 
-# TODO: implement domain leave actions
-# action :leave do
-#
-# end
+action :leave do
+    
+  reboot 'Restart Computer' do
+    action :nothing
+  end
+
+  case node['os']
+  when 'windows'
+    warning_caption = 'Chef is leaving the domain'
+    warning_text = 'The chef cookbook ad-join is currently in the middle of leaving the domain, and the server is about to be restarted'
+    warning_key = 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System'
+
+    # Display a warning incase anyone remotes into the machine before all reboots have finished
+    # http://www.techrepublic.com/blog/windows-and-office/adding-messages-to-windows-7s-logon-screen/
+    registry_key 'warning' do
+      key warning_key
+      values [
+        { name: 'legalnoticecaption', type: :string, data: warning_caption },
+        { name: 'legalnoticetext', type: :string, data: warning_text }
+      ]
+      only_if { node['ad-join']['windows']['visual_warning'] == true }
+      action :nothing
```

Comment:
I liked the way this was implemented in the join action. The key only gets created if there is any action needed on leaving the domain (e.g. why create the message if we are already not on a domain?).